lib/Marlin/Marlin/src/feature/prusa/MMU2/protocol_logic.cpp

```diff
@@ -737,6 +737,8 @@ void ProtocolLogic::FormatLastResponseMsgAndClearLRB(char *dst) {
 
 void ProtocolLogic::LogRequestMsg(const uint8_t *txbuff, uint8_t size) {
     constexpr uint_fast8_t rqs = modules::protocol::Protocol::MaxRequestSize() + 1;
+    assert((size + 1) < (uint8_t)rqs);
```

Comment:
Is it really possible for `(size+1 >= rqs)` to happen? If yes and if it cannot be checked at compile time, the correct spot for this check are unit tests in the Prusa-Firmware-MMU repo.

Reply:
Yes, should be added to the unit tests then, or better get rid completely of this function / code.

Reply:
Logging is an important part of debugging, it must not be removed.

Reply:
the assert is now added to the unit tests.